allen-cell-animated/simularium-website
# Feature/migrate Mocha/Chai to Jest

Problem
=======
Resolves #179 

Solution
========
1. I referred to a combination of Jest docs, the current simularium-viewer setup, and various sources from the Internet to reconfigure our npm packages so that Jest could find and run our tests properly.
    
    From Jest home page:
    > **zero config**
    > Jest aims to work out of the box, config free, on most JavaScript projects.

    Lies. Oh wait key word "aims" 😄 


2. I used the [jest-codemods tool](https://jestjs.io/docs/migration-guide#jest-codemods) to automatically convert most of the Chai syntax to Jest. It successfully converted ~40 out of 44 total tests which is pretty impressive 🤷🏻‍♀️ 
3. I removed Mocha and Chai from package.json and any config files for them

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)
* This change requires updated or new tests

Steps to Verify:
----------------
1. Pull this branch and run `npm run test`, check that all 44 tests are run with Jest and that all of them pass
2. Also run `npm start` to make sure other things aren't broken 

Screenshots (optional):
-----------------------
![image](https://user-images.githubusercontent.com/12690133/125704594-92c64f58-de88-483b-8f22-e5bb2b5aca62.png)
